pkg/asset/installconfig/azure/validation.go

```diff
@@ -215,12 +232,24 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 			if ultraSSDCapability == "" {
 				ultraSSDCapability = defaultUltraSSDCapability
 			}
+			capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+			}
 			if vmNetworkingType == "" {
-				vmNetworkingType = defaultVMNetworkingType
+				val, ok := capabilities[string(azure.AcceleratedNetworkingEnabled)]
+				if ok {
+					if strings.EqualFold(val, "True") {
+						vmNetworkingType = string(azure.VMnetworkingTypeAccelerated)
+					} else {
+						vmNetworkingType = string(azure.VMNetworkingTypeBasic)
+					}
+				}
 			}
+			ic.Compute[idx].Platform.Azure.VMNetworkingType = vmNetworkingType
```

Comment:
Same comment as above, I don't think we need to set defaults in the validation for this.